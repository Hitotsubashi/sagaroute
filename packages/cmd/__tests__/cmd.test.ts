import { exec } from 'child_process';
import fs from 'fs';
import path from 'path';

test('use cmd', (done) => {
  exec('cd ./__tests__/ignore-test-normal && sagaroute', (error) => {
    if (error) {
      done(error);
      return;
    }
    const context = fs.readFileSync(
      path.join(__dirname, 'ignore-test-normal', 'src', 'routes.tsx'),
      'utf-8',
    );
    expect(context).toMatchSnapshot();
    done();
  });
});

test('use cmd with specified routeFilePath and lazy ', (done) => {
  exec(
    'cd ./__tests__/ignore-test-normal && sagaroute --routeFilePath=./src/routes1.tsx --lazy=true',
    (error) => {
      if (error) {
        done(error);
        return;
      }
      const context = fs.readFileSync(
        path.join(__dirname, 'ignore-test-normal', 'src', 'routes1.tsx'),
        'utf-8',
      );
      expect(context).toMatchSnapshot();
      done();
    },
  );
});

test('use cmd with specified dirpath and layoutDirPath ', (done) => {
  exec(
    'cd ./__tests__/ignore-test-dirpath && sagaroute --dirpath=src/views --layoutDirPath=./src/dashboard',
    (error) => {
      if (error) {
        done(error);
        return;
      }
      const context = fs.readFileSync(
        path.join(__dirname, 'ignore-test-dirpath', 'src', 'routes.tsx'),
        'utf-8',
      );
      expect(context).toMatchSnapshot();
      done();
    },
  );
});

test('use cmd with specified error dirpath', (done) => {
  exec('cd ./__tests__/ignore-test-dirpath && sagaroute --dirpath=./src/pages', (error) => {
    expect(error?.message).toEqual(
      expect.stringContaining(
        `Cannot find folder with path "${path.join(
          __dirname,
          '/ignore-test-dirpath',
          './src/pages',
        )}"`,
      ),
    );
    done();
  });
});

test('use cmd with warnings include lost variable in both view and template', (done) => {
  exec('cd ./__tests__/ignore-test-warning && sagaroute', (error, stdout) => {
    if (error) {
      done(error);
      return;
    }
    const problem1 = `1. The view of stage<print.inject.before> has no variables such as [prefix] which are needed in template.`;
    expect(stdout).toEqual(
      expect.stringContaining(
        `sagaroute has completed the work, but there are the following problems:\n${problem1}`,
      ),
    );
    done();
  });
});
