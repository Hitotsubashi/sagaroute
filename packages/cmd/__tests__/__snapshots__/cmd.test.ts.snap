// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use cmd 1`] = `
"// @ts-nocheck
/* sagaroute-inject:imports */
/* injected by sagaroute: start */
import LayoutsIndex from "./layouts/index.tsx";
import PagesIndex from "./pages/index.tsx";
import Pages404 from "./pages/404.tsx";
/* injected by sagaroute: end */

// sagaroute-inject: routes
const routes = [
  {
    "path": "/",
    "element": <LayoutsIndex/>,
    "children": [
      {
        "index": true,
        "element": <PagesIndex/>
      }
    ]
  },
  {
    "path": "*",
    "element": <Pages404/>
  }
];
export default routes;"
`;

exports[`use cmd with specified dirpath and layoutDirPath  1`] = `
"// @ts-nocheck
/* sagaroute-inject:imports */
/* injected by sagaroute: start */
import DashboardIndex from "./dashboard/index.tsx";
import ViewsIndex from "./views/index.tsx";
/* injected by sagaroute: end */

// sagaroute-inject: routes
const routes = [
  {
    "path": "/",
    "element": <DashboardIndex/>,
    "children": [
      {
        "index": true,
        "element": <ViewsIndex/>
      }
    ]
  }
];
export default routes;"
`;

exports[`use cmd with specified routeFilePath and lazy  1`] = `
"// @ts-nocheck
/* sagaroute-inject:imports */

// sagaroute-inject: routes
const routes = [
  {
    "path": "/",
    "lazy": async function() {
    const {default : LayoutsIndex}  = await import("./layouts/index.tsx");
    return {
  "Component": LayoutsIndex
};
  },
    "children": [
      {
        "index": true,
        "lazy": async function() {
    const {default : PagesIndex}  = await import("./pages/index.tsx");
    return {
  "Component": PagesIndex
};
  }
      }
    ]
  },
  {
    "path": "*",
    "lazy": async function() {
    const {default : Pages404}  = await import("./pages/404.tsx");
    return {
  "Component": Pages404
};
  }
  }
];
export default routes;"
`;
